src/misc/useOffset.ts

```diff
@@ -0,0 +1,26 @@
+import { useState } from 'react';
+
+import useSearchParams from './useSearchParams';
+
+export const limit = 21;
+export default function useOffset(location: Location, keyName = 'offset') {
+  const [searchParams, setSearchParams] = useSearchParams(location);
+  const offset = keyName in searchParams ? searchParams[keyName] : 0;
+  const setOffset = (offset: number) => {
+    if (offset < 1) {
+      const newSearchParams = { ...searchParams };
+      delete newSearchParams[keyName];
```

Comment:
I wonder why it deletes after copy `searchParams`, not delete directly from `searchParams`.